defs/go.bzl

```diff
@@ -70,7 +66,7 @@ def _go_genrule_impl(ctx):
   ctx.action(
       inputs = list(all_srcs) + resolved_inputs,
       outputs = ctx.outputs.outs,
-      env = ctx.configuration.default_shell_env,
+      env = ctx.configuration.default_shell_env + ctx.attr._go_toolchain.env,
```

Comment:
stuff like this came from https://github.com/bazelbuild/rules_go/blob/master/go/private/go_toolchain.bzl#L51-L71